src/emulator/io/include/io/file_wrapper.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+#include <string>
+
+/// <summary>
+/// Class that aims to encapsulate operations related to file manipulation, independent from any OS
+/// </summary>
+class FileWrapper {
+public:
+    /// <summary>
+    /// wIndicates if the file is currently opened.
+    /// </summary>
+    bool file_open = false;
+
+    /// <summary>
+    /// Opens the file.
+    /// </summary>
+    /// <param name="path">The path.</param>
+    /// <param name="read_mode">The read mode.</param>
+    /// <returns>Integer representing if the operation was successful</returns>
+    int open_file(std::string path, std::string read_mode);
```

Comment:
Remind you in the future, if you are not going to modify the argument string, passed it as a const reference

```suggestion
    int open_file(const std::string &path, std::string read_mode);
```